### DISCUSSION TITLE
Consider migrating to Actix-Web

### DISCUSSION THREAD

**Discussion by nyurik** [maplibre/mbtileserver-rs#17] [Ideas]
Hyper is a fairly low level http library. I think it might make sense to migrate to Actix-Web that supports more advanced functionality like TLS. Plus it seems Actix-Web has a bigger community and it is more active.

**Comment by maxammann** [comment]
I think using any HTTP server framework is fine as long as it offers the features we need. But we should not expose any configuration which is specific to actix or hyper, such that we can easily change the framework in the future if we want to.

**Comment by ka7eh** [comment]
I second @maxammann comment.

I've been thinking of adding TLS support, but I haven't had time to look into it. If doing it with Actix-Web is more trivial than hyper, then I'm fine with switching to Actix-Web.
